scripts/generate-abi.py

```diff
@@ -0,0 +1,87 @@
+import json
+import tempfile
+from pathlib import Path
+
+import click
+from psutil import Popen
+
+_COMPILE_COMMAND = """
+git clone --no-checkout git@github.com:beamer-bridge/beamer.git {beamer_path} &&
+cd {beamer_path} &&
+git checkout {commit_id} &&
+ape compile
+"""
+
+
+def _get_contract_commit_ids(deployment_path: Path) -> dict[str, set[str]]:
+    file_contract_commits: dict[str, set[str]] = {}
+    for file_path in deployment_path.glob("*.deployment.json"):
+        with open(file_path, "r") as fp:
+            data = json.load(fp)
+        data.pop("deployer")
+        for contract_name, contract_info in data.items():
+            if contract_name not in file_contract_commits:
+                file_contract_commits[contract_name] = set()
+            file_contract_commits[contract_name].add(contract_info["beamer_commit"])
+    return file_contract_commits
+
+
+def _compile_abis(commit_id: str, beamer_path: str) -> None:
+    command = _COMPILE_COMMAND.format(beamer_path=beamer_path, commit_id=commit_id).replace(
+        "\n", ""
+    )
+    Popen(command, shell=True).wait()
+
+
+def _merge_abis(generated_abis: dict[str, dict[str, str]]) -> dict[str, str]:
+    merged_abis = {}
+
+    for contract_name, commit_abis in generated_abis.items():
+        if len(set(commit_abis.values())) > 1:
+            raise ValueError(f"Different ABI found for {contract_name}")
+        merged_abis[contract_name] = next(iter(commit_abis.values()))
+
+    return merged_abis
+
+
+def _generate_abis(contract_commit_map: dict[str, set[str]]) -> dict[str, dict[str, str]]:
+    generated_abis: dict[str, dict[str, str]] = {}
+    processed_commit_ids = []
+
+    for commit_ids in contract_commit_map.values():
+        for commit_id in commit_ids:
+            if commit_id in processed_commit_ids:
```

Comment:
This is going to do a linear search so I suggest making `processed_commit_ids` a `set` to turn this into a constant-time operation.